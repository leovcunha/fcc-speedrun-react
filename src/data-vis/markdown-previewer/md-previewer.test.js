import React from "react";
import renderer from "react-test-renderer";
import MDPreviewer from "../markdown-previewer";

describe("md-previewer", () => {
    it("renders without crashing", () => {
        const component = renderer.create(<MDPreviewer/>).toJSON();
        expect(component).toMatchSnapshot();
    });
});
