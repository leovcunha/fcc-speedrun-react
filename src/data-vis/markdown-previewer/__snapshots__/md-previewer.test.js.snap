// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`md-previewer renders without crashing 1`] = `
<div
  className="md-previewer jumbotron"
>
  <div
    className="row h1 justify-content-center"
  >
    Markdown Previewer
  </div>
  <hr
    className="p-3"
  />
  <div
    className="row"
  >
    <div
      className="col-sm-6"
    >
      <textarea
        onChange={[Function]}
        value="Heading
=======

Sub-heading
-----------
 
### Another deeper heading
 
Paragraphs are separated
by a blank line.

Leave 2 spaces at the end of a line to do a  
line break

Text attributes *italic*, **bold**, 
\`monospace\`, ~~strikethrough~~ .

Shopping list:

  * apples
  * oranges
  * pears

Numbered list:

  1. apples
  2. oranges
  3. pears

"
      />
    </div>
    <div
      className="col-sm-6"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h1 id=\\"heading\\">Heading</h1>
<h2 id=\\"sub-heading\\">Sub-heading</h2>
<h3 id=\\"another-deeper-heading\\">Another deeper heading</h3>
<p>Paragraphs are separated
by a blank line.</p>
<p>Leave 2 spaces at the end of a line to do a<br>line break</p>
<p>Text attributes <em>italic</em>, <strong>bold</strong>, 
<code>monospace</code>, <del>strikethrough</del> .</p>
<p>Shopping list:</p>
<ul>
<li>apples</li>
<li>oranges</li>
<li>pears</li>
</ul>
<p>Numbered list:</p>
<ol>
<li>apples</li>
<li>oranges</li>
<li>pears</li>
</ol>
",
        }
      }
    />
  </div>
</div>
`;
